### PR TITLE
add access control

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

**Comment by gregLibert** [comment]
see massalabs/massa-standards#153

**Comment by gregLibert** [comment]
@Ben-Rey can you with the CI?

Furthermore, I decided to create a `security` module because many functions (evmGetPubkeyFromSignature, isEvmSignatureValid, isSignatureValid, keccak256, sha256...) could be removed from utils to improve clarity.
Wdyt?

**Comment by Ben-Rey** [comment]
OK perfect, I'll check the ci

File: assembly/__tests__/access_control.spec.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    const USER = accessControl.newPermission('user');
+    const GUEST = accessControl.newPermission('guest');
+    const userAddress = new Address(
+      'AU12UBnqTHDQALpocVBnkPNy7y5CndUJQTLutaVDDFgMJcq5kQiKq',
+    );
+
+    // accessControl.mustHavePermission(ADMIN, userAddress);
+    // accessControl.mustHavePermission(USER, userAddress);
+    // accessControl.mustHavePermission(GUEST, userAddress);
+  });
```
Comment (thomas-senechal) [inline assembly/__tests__/access_control.spec.ts:236]: Is this test useful ?

**Review by thomas-senechal — APPROVED** [review]

File: assembly/__tests__/access_control.spec.ts (hunk 1 above)
Comment (gregLibert) [reply assembly/__tests__/access_control.spec.ts:236]: Yes, it let me manually check that the assert message is the correct one. To be remove once this test would be performed by an integration test.

**Review by Ben-Rey — APPROVED** [review]

**Pull request merged**